index.html

```diff
@@ -341,6 +420,19 @@ <h2>
         following algorithm:
       </p>
       <ol>
+        <li>
+          Let <var>p</var> be a new <a>Promise</a> object.
+        </li>
+        <li>
+          Run the following steps <a>in parallel</a>:
```

Comment:
Doing this in parallel means the UA could finish the serial steps and fulfil _p_ before getting around to checking for permission.